enterprise/server/remote_execution/execution_server/execution_server.go

```diff
@@ -1090,13 +1125,33 @@ func (s *ExecutionServer) cacheExecuteResponse(ctx context.Context, taskID strin
 	return cachetools.UploadActionResult(ctx, s.env.GetActionCacheClient(), arn, ar)
 }
 
+func (s *ExecutionServer) cacheActionResult(ctx context.Context, actionResourceName *digest.ResourceName, response *repb.ExecuteResponse, md *espb.ExecutionAuxiliaryMetadata) error {
+	if response.GetCachedResult() || md.GetExecutionTask() == nil {
+		return nil
+	}
+	// The action executed and the executor sent us an ExecutionTask in the
+	// auxiliary metadata, which indicates we need to write to the cache.
+	action := md.GetExecutionTask().GetAction()
+	// If the action failed or do_not_cache is set, upload information about the error via a failed
+	// ActionResult under an invocation-specific digest, which will not ever be seen by bazel but
+	// may be viewed via the Buildbuddy UI.
+	cacheableResourceName := actionResourceName
+	if action.GetDoNotCache() || response.GetStatus().GetCode() != 0 || response.GetResult().GetExitCode() != 0 {
+		resultDigest, err := digest.AddInvocationIDToDigest(actionResourceName.GetDigest(), actionResourceName.GetDigestFunction(), md.GetExecutionTask().GetInvocationId())
+		if err != nil {
+			return status.UnavailableErrorf("Error uploading action result: %s", err.Error())
+		}
+		cacheableResourceName = digest.NewResourceName(resultDigest, actionResourceName.GetInstanceName(), rspb.CacheType_AC, actionResourceName.GetDigestFunction())
```

Comment:
I think these failed ActionResults are no longer used, so we can short-circuit in this case. The UI fetches the response that was stored with cacheExecuteResponse now, which includes the complete response and works for both failed and succeeded actions.

Reply:
I sent https://github.com/buildbuddy-io/buildbuddy/pull/7985 to first remove these writes.